src/client/wieldmesh.cpp

```diff
@@ -432,19 +430,17 @@ void WieldMeshSceneNode::setItem(const ItemStack &item, Client *client, bool che
 			material.setFlag(video::EMF_BILINEAR_FILTER, m_bilinear_filter);
 			material.setFlag(video::EMF_TRILINEAR_FILTER, m_trilinear_filter);
 		}
-		return;
 	}
 	else if (!def.inventory_image.empty()) {
 		setExtruded(def.inventory_image, def.inventory_overlay, def.wield_scale,
 			tsrc, 1);
 		m_colors.emplace_back();
 		// overlay is white, if present
 		m_colors.emplace_back(true, video::SColor(0xFFFFFFFF));
-		return;
 	}
 
-	// no wield mesh found
-	changeToMesh(nullptr);
```

Comment:
This call should be kept to properly update the mesh when none of the cases above matches.
```C++
	m_colors.emplace_back(true, video::SColor(0xFFFFFFFF));
} else {
	// no wield mesh found
	changeToMesh(nullptr);
}
```

EDIT: Only works if all cases use the same `else if` chain.

Reply:
Yessir, gotit.